Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs

```diff
@@ -228,8 +228,17 @@ public void ResetConfiguration(MixedRealityToolkitConfigurationProfile profile)
             if (IsCoreSystem(interfaceType))
             {
                 activeSystems.Remove(interfaceType);
-                // Core services are always removable
                 MixedRealityServiceRegistry.RemoveService<T>(serviceInstance, this);
+
+                // Reset the convenience properties.
+                if (typeof(IMixedRealityBoundarySystem).IsAssignableFrom(interfaceType)) { boundarySystem = null; }
+                else if (typeof(IMixedRealityCameraSystem).IsAssignableFrom(interfaceType)) { cameraSystem = null; }
+                else if (typeof(IMixedRealityDiagnosticsSystem).IsAssignableFrom(interfaceType)) { diagnosticsSystem = null; }
+                // Focus provider reference is not managed by the MixedRealityToolkit class.
```

Comment:
Should we clear the input system's reference to the focus provider?

Reply:
Should that happen in the input system itself on its own destroy call?

---
In reply to: [288768978](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/4619#discussion_r288768978) [](ancestors = 288768978)

Reply:
This question leads to a longer discussion of why the input system owns the reference to the focus provider, instead of the focus provider being an independently registered system.

Reply:
changing the service locator to track the focus provider is too large of a change at this time. also, the input system interface makes the focus provider a read-only property and we are no longer taking breaking changes.

we can revisit how the focus provider is managed in a future release